packages/main/src/SliderBase.js

```diff
@@ -0,0 +1,623 @@
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.main.SliderBase
+ * @extends sap.ui.webcomponents.base.UI5Element
+ * @tagname ui5-slider
+ * @public
+ */
+class SliderBase extends UI5Element {
+	constructor() {
+		super();
+		this._resizeHandler = this._handleResize.bind(this);
+		this._moveHandler = this._handleMove.bind(this);
+		this._upHandler = this._handleUp.bind(this);
+
+		this.TICKMARK_COLOR_MAP = {
+			sap_fiori_3: "#89919a",
+			sap_fiori_3_dark: "#89919a",
+			sap_fiori_3_hcw: "#000000",
+			sap_fiori_3_hcb: "#ffffff",
+			sap_belize: "#bfbfbf",
+			sap_belize_hcw: "#000000",
+			sap_belize_hcb: "#ffffff",
+		};
+
+		this._stateStorage = {
+			step: null,
+			min: null,
+			max: null,
+		};
+	}
+
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return styles;
+	}
+
+	static get UP_EVENTS() {
+		return ["mouseup", "pointerup", "touchend"];
+	}
+
+	static get MOVE_EVENT_MAP() {
+		return {
+			mousedown: "mousemove",
+			pointerdown: "pointermove",
+			touchstart: "touchmove",
+		};
+	}
+
+	get labelItems() {
+		return this._labelItems;
+	}
+
+	get classes() {
+		return {
+			labelContainer: {
+				"ui5-slider-hidden-labels": this._labelsOverlapping,
+			},
+		};
+	}
+
+	onEnterDOM() {
+		ResizeHandler.register(this, this._resizeHandler);
+		this.addEventListener("mouseover", this._mouseOverHandler);
+		this.addEventListener("mouseout", this._mouseOutHandler);
+	}
+
+	onExitDOM() {
+		ResizeHandler.deregister(this, this._handleResize);
+		this.removeEventListener("mouseover", this._mouseOverHandler);
+		this.removeEventListener("mouseout", this._mouseOutHandler);
+	}
+
+	onAfterRendering() {
+		// Only call if the resize is triggered by a state changes other than
+		// the ones that occured on the previous resize and those caused by user interaction.
+		if (this.notResized) {
+			this._resizeHandler();
+		}
+	}
+
+	_ontouchstart(event) {
+		this._onmousedown(event);
+	}
+
+	_ontpointerdown(event) {
+		this._onmousedown(event);
+	}
+
+	/** Shows the tooltip(s) if the <code>showTooltip</code> property is set to true
+	 *
+	 * @private
+	 */
+	_onmouseover(event) {
+		if (!this.disabled || this.showTooltip) {
+			this._tooltipVisibility = "visible";
+		}
+	}
+
+	/**
+	 * Hides the tooltip(s) if the <code>showTooltip</code> property is set to true
+	 *
+	 * @private
+	 */
+	_onmouseout(event) {
+		if (this.showTooltip) {
+			this._tooltipVisibility = "hidden";
+		}
+	}
+
+	/**
+	 * Handle the responsiveness of the Slider's UI elements when resing
+	 *
+	 * @private
+	 */
+	_handleResize() {
+		if (!this.tickmarks) {
+			return;
+		}
+
+		// Mark resizing to avoid unneccessary calls to that function after rendering
+		this.notResized = false;
+
+		// Convert the string represented calculation expression to a normal one
+		// Check the distance  in pixels exist between every tickmark
+		const tickmarksAmountStrCalc = this._tickmarksAmount.split("/");
```

Comment:
this can be extracted in a method to improve the readability of _handleResize
```js
const tickmarksAmountStrCalc = this._tickmarksAmount.split("/");
		const tickmarksAmount = tickmarksAmountStrCalc[0] / tickmarksAmountStrCalc[1];
		const spaceBetweenTickmarks = 
```